### PR TITLE
DRAFT: feat: make the Appium Inspector openable from command line

### PR DESCRIPTION
**Pull request by dpgraham**
Add a feature where you can open Appium Inspector from command line:

Example:

```
node . --session-file ./sample-session-files/fake.appiumsession
```

This will open the inspector in the user's default browser and populate the state with the contents of the session file.

TODO: This will require a future pull request that publishes this as an NPM package.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+const open = require('open');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers')
+const { fs } = require('appium-support');
```
Comment (mykola-mokhnach) [inline main.js:4]: this module is not supported anymore. Consider switching to `@appium/support` one

File: main.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    } catch (e) {
+      throw new Error(`Session file ${sessionFile} is not valid JSON`);
+    }
+  }
+
+  if (stateJson) {
+    try {
+      state = {...state, ...JSON.parse(stateJson)};
+    } catch (e) {
+      throw new Error(`State JSON is not valid JSON`);
```
Comment (mykola-mokhnach) [inline main.js:44]: strings or arrays are also valid JSON

File: main.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+module.exports.openAppiumInspector = async function openAppiumInspector ({
+  sessionFile, stateJson, autoStart = false,
+}) {
+  // start a static server to serve the dist-browser files
+  const distBrowser = path.join(__dirname, 'dist-browser');
+  const serve = serveStatic(distBrowser, {index: ['index.html']});
+  const server = http.createServer(function onRequest (req, res) {
+    serve(req, res, finalhandler)
+  });
+  const port = await portfinder.getPortPromise({port: 8888});
```
Comment (mykola-mokhnach) [inline main.js:23]: should port number be configurable?

**Comment by jlipps** [comment]
seems like this is including a lot of irrelevant commits and should be rebased on master.

also i'm not entirely sure of the purpose of this change. can't the inspector already be started from the command line (`open Inspector.app`?)

**Comment by dpgraham** [comment]
This is just a draft. The purpose of this is so that you can open the inspector from a client. Something like webdriverio could call `debug` and then that would open the running session in the inspector.

**Pull request closed without merging**